Add zstd support for llvm 16

SaltyKitkat opened the pull request:
###### Description of changes

LLVM add zstd support since 16

And maybe we should support that.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

pbsds committed:
code-server: add vscode-with-extensions compatibility

pbsds committed:
curv: init at 0.5

Commit:
maintainers: add hulr

Commit:
python310Packages.pytest-testinfra: init at 8.1.0

https://github.com/pytest-dev/pytest-testinfra/releases/tag/8.1.0

lopsided98 committed:
urdfdom: cleanup dependencies

* urdfdom no longer depends on boost
* urdfdom includes tinyxml.h in its public headers, therefore tinyxml must be
  propagated

DerDennisOP committed:
python3.pkgs.django-rosetta: init at 0.9.8

MagicRB committed:
nomad_1_5: init

Signed-off-by: Magic_RB <magic_rb@redalder.org>

DerDennisOP committed:
python3.pkgs.django-admin-datta: init at 1.0.7

e1mo committed:
python3Packages.python-ipware: init at 0.9.0

e1mo committed:
paperless-ngx: 1.14.4 -> 1.16.1

Various smaller new features (quick filters from document detail,
explanations for relative dates, pagination for tasks) and various other
bug fixes and minor enhancements. v1.16.0 contains seemingly smaller
features and and changed their tika client libary.
Especially noteworthy for v1.14.5 seems a security fix around the display
of texts in the frontend.

Full changelog available at
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.14.5
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.15.0
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.15.1
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.16.0
https://github.com/paperless-ngx/paperless-ngx/releases/tag/v1.16.1

markuskowa committed:
hdf5: 1.14.0 -> 1.14.1-2

f2k1de committed:
graylogPlugins.smseagle: init at 1.0.1

xworld21 committed:
texlive.tlpdb.nix: better sed looping technique

xworld21 committed:
texlive.tlpdb.nix: do not escape valid names

StarrFox committed:
imhex: 1.27.1 -> 1.29.0

r-vdp committed:
fwupd: 1.8.15 -> 1.9.1

Release notes: https://github.com/fwupd/fwupd/releases/tag/1.9.1

Commit:
Fix: make Espanso.app on darwin

This change creates an app on darwin, as this is expected on the
platform.

olivia-fl committed:
maintainers: add Benjamin-L

olivia-fl committed:
deqp-runner: init at 0.16.1

fabaff committed:
python311Packages.google-api-python-client: 2.84.0 -> 2.88.0

Changelog: https://github.com/googleapis/google-api-python-client/releases/tag/v2.88.0

Janik-Haag committed:
python3Packages.django-leaflet: init at 0.29.0

GaetanLepage committed:
python3Packages.mypy: 1.0.1 -> 1.3.0

GaetanLepage committed:
python3Packages.mypy-extensions: 0.4.3 -> 1.0.0

CaptainJawZ committed:
furtherance: 1.7.0 -> 1.8.0

SuperSandro2000 committed:
python310Packages.aiokef: remove linters and coverage collecting

SuperSandro2000 committed:
python310Packages.returns: remove linters, coverage, update homepage

SuperSandro2000 committed:
python310Packages.pixelmatch: remove mypy linter

chvp committed:
udp2raw: disable address sanitization

udp2raw crashes very frequently when it's turned on. Reported upstream at
https://github.com/wangyu-/udp2raw/issues/474/.

SuperSandro2000 committed:
python310Packages.loguru: remove mypy linter

SuperSandro2000 committed:
python310Packages.python-utils: remove mypy linter

SuperSandro2000 committed:
python310Packages.loguru: disable test which fails on some machines

SuperSandro2000 committed:
python310Packages.magicgui: remove mypy linter

SuperSandro2000 committed:
python310Packages.psygnal: remove mypy linter

SuperSandro2000 committed:
python310Packages.portalocker: remove mypy linter, remove default meta.platforms

SuperSandro2000 committed:
nix-pin: remove mypy linter

SuperSandro2000 committed:
python310Packages.strawberry-graphql: remove mypy linter

SuperSandro2000 committed:
ulauncher: remove mypy linter

SuperSandro2000 committed:
python310Packages.dataclasses-json: remove mypy linter

SuperSandro2000 committed:
python310Packages.duckdb: remove mypy linter

SuperSandro2000 committed:
python310Packages.fingerprints: fix test execution

SuperSandro2000 committed:
python310Packages.globus-sdk: remove unused mypy linter

SuperSandro2000 committed:
python310Packages.immutables: remove mypy linter

SuperSandro2000 committed:
python310Packages.nptyping: remove mypy linter

SuperSandro2000 committed:
python310Packages.plugwise: remove mypy linter

SuperSandro2000 committed:
python310Packages.typical: remove mypy linter

SuperSandro2000 committed:
zfs-replicate: remove useless pytest-runner, coverage collecting and mypy linter

SuperSandro2000 committed:
python310Packages.qtile: remove mypyi linter

CaptainJawZ committed:
furtherance 1.8.0 changed rec into finalAttrs

ners committed:
mermaid-filter: init at 1.4.6

figsoda committed:
typst: 0.4.0 -> 0.5.0

Diff: https://github.com/typst/typst/compare/v0.4.0...v0.5.0

Changelog: https://github.com/typst/typst/releases/tag/v0.5.0

figsoda committed:
ruff: 0.0.270 -> 0.0.272

Diff: https://github.com/astral-sh/ruff/compare/v0.0.270...v0.0.272

Changelog: https://github.com/astral-sh/ruff/releases/tag/v0.0.272

lilyinstarlight committed:
gst_all_1.gst-plugins-rs: 0.10.7 -> 0.10.8

OPNA2608 committed:
libsidplayfp: 2.4.2 -> 2.5.0

foo-dogsquared committed:
halftone: 0.2.1 -> 0.3.0

AndersonTorres committed:
cmake: 3.25.3 -> 3.26.4

GaetanLepage committed:
python310Packages.tatsu: remove mypy linter

GetPsyched committed:
maintainers: add getpsyched

GetPsyched committed:
atlauncher: init at 3.4.28.1

r-vdp committed:
fwupd: 1.9.1 -> 1.9.2

Release notes: https://github.com/fwupd/fwupd/releases/tag/1.9.2

dependabot[bot] committed:
build(deps): bump korthout/backport-action from 1.2.0 to 1.3.1

Bumps [korthout/backport-action](https://github.com/korthout/backport-action) from 1.2.0 to 1.3.1.
- [Release notes](https://github.com/korthout/backport-action/releases)
- [Commits](https://github.com/korthout/backport-action/compare/v1.2.0...v1.3.1)

---
updated-dependencies:
- dependency-name: korthout/backport-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

SFrijters committed:
python3Packages.ziamath: 0.7 -> 0.8.1

SFrijters committed:
python3Packages.schemdraw: 0.16 -> 0.17

figsoda committed:
erdtree: 3.0.1 -> 3.0.2

Diff: https://github.com/solidiquis/erdtree/compare/v3.0.1...v3.0.2

Changelog: https://github.com/solidiquis/erdtree/releases/tag/v3.0.2

ncfavier committed:
runCommand: set meta.position from the arguments

Currently all packages defined by a call to runCommand have their
meta.position pointing to trivial-builders.nix.

This change makes uses of runCommand that supply at least one attribute
in the second argument point to the right file.

SFrijters committed:
maestral-qt: 1.7.2 -> 1.7.3

tfmoraes committed:
blackbox-terminal: 0.13.1 -> 0.13.2

bcdarwin committed:
python310Packages.summarytools: init at 0.2.3

tie committed:
rubyPackages.pg: use pkg-config instead of pg_config

This change forces pg native extension to use pkg-config to locate libpq
instead of running pg_config. It also removes an unnecessary reference
to postgresql package since build flags are embedded in the output at
`$out/${ruby.gemPath}/build_info/*`.

fabianhjr committed:
lib.systems: add znver4 architecture

ShamrockLee committed:
losslesscut-bin: 3.48.2 -> 3.55.2

ShamrockLee committed:
losslesscut-bin.x86_64-darwin: change platform to platforms.darwin

emilytrau committed:
python3Packages.ev3dev2: fix build failure

SuperSandro2000 committed:
python310Packages.zimports: disable on python 3.11

SuperSandro2000 committed:
refurb: disable test relying on mypy

emilytrau committed:
python3Packages.python-ev3dev2: rename from python3Packages.ev3dev2

mdarocha committed:
openraPackages_2019: fix build
All mods and engines were failing to build due to path issues
after package name changes.

figsoda committed:
typst-lsp: 0.5.1 -> 0.6.2

Diff: https://github.com/nvarner/typst-lsp/compare/v0.5.1...v0.6.2

Changelog: https://github.com/nvarner/typst-lsp/releases/tag/v0.6.2

e1mo committed:
python3Packages.tika-client: init at 0.1.0

nagy committed:
gforth: 0.7.9_20220127 -> 0.7.9_20230518

Diff: https://github.com/forthy42/gforth/compare/0.7.9_20220127...0.7.9_20230518

SuperSandro2000 committed:
moon: change version to confirm to contributing guide

natsukium committed:
python3Packages.meeko: init at 0.4.0

marsam committed:
python310Packages.pydantic: 1.10.8 -> 1.10.9

Diff: https://github.com/pydantic/pydantic/compare/refs/tags/v1.10.8...v1.10.9

Changelog: https://github.com/pydantic/pydantic/blob/v1.10.9/HISTORY.md

marsam committed:
python310Packages.python-utils: 3.5.2 -> 3.6.0

Diff: https://github.com/WoLpH/python-utils/compare/refs/tags/v3.5.2...v3.6.0

Changelog: https://github.com/wolph/python-utils/releases/tag/v3.6.0

fabaff committed:
python311Packages.emoji: 2.5.0 -> 2.5.1

Diff: https://github.com/carpedm20/emoji/compare/refs/tags/v2.5.0...v2.5.1

Changelog: https://github.com/carpedm20/emoji/blob/v2.5.1/CHANGES.md

nyanotech committed:
python3Packages.pem: Init at 21.2.0

nyanotech committed:
syncplay: fix tls support in client

Adds "pem", which is a new dependency added in the last release from upstream.

Nixos patches certifi to return the system ca bundle, which includes openssl-format "trusted certificate"s, which pyopenssl seems to choke on when syncplay tries to load them. Therefore, we add a patch that skips those "trusted certificates".

pbek committed:
qownnotes: 23.6.4 -> 23.6.5

fabaff committed:
creds: init at 0.5

Tillerino committed:
dbeaver: use JDK17 for Maven build

JDK17 was already used to run DBeaver, but the Maven build started
failing recently when the JDK used implicitly by Maven was upgraded.

figsoda committed:
cargo-tarpaulin: 0.25.2 -> 0.26.0

Diff: https://github.com/xd009642/tarpaulin/compare/0.25.2...0.26.0

Changelog: https://github.com/xd009642/tarpaulin/blob/0.26.0/CHANGELOG.md

samhug committed:
ntopng: fix missing static assets

Instead of building the `dist` make target in the derivation (requires npm and
all that), we're using the upstream pre-built static assets. Upstream refers to
these using a sub-module so we need to fetch that.

Without this patch the web UI loads the html, but 404s on all CSS and JS assets
because the `$out/share/ntopng/httpdocs/dist` directory is empty

r-ryantm committed:
onlyoffice-documentserver: 7.3.3 -> 7.4.0

MatthewCash committed:
nixos/kmscon: disable reload-systemd-vconsole-setup service

marsam committed:
gallery-dl: 1.25.5 -> 1.25.6

Changelog: https://github.com/mikf/gallery-dl/blob/v1.25.6/CHANGELOG.md

marsam committed:
python310Packages.nvchecker: 2.11 -> 2.12

Diff: https://github.com/lilydjwg/nvchecker/compare/v2.11...v2.12

Changelog: https://github.com/lilydjwg/nvchecker/releases/tag/v2.12

r-ryantm committed:
mmc-utils: unstable-2023-04-17 -> unstable-2023-06-12

kirillrdy committed:
python3Packages.howdoi: 2.0.19 -> 2.0.20 enable more tests, unmark
broken on aarch64-linux

Yarny0 committed:
tsm-client: 8.1.17.2 -> 8.1.19.0

"Update History" (release notes):
https://www.ibm.com/support/pages/node/6998343

With this release, IBM renamed the product from
"IBM Spectrum Protect" to "IBM Storage Protect":
https://www.ibm.com/support/pages/node/6964770
However, URLs apparently haven't been changed (yet).

fabaff committed:
nbutools: init at unstable-2023-06-06

meppu committed:
gigalixir: 1.3.0 -> 1.6.0

fabaff committed:
python311Packages.mypy-boto3-s3: 1.26.127 -> 1.26.155

fabaff committed:
python311Packages.pysigma-backend-insightidr: 0.1.9 -> 0.2.1

Diff: https://github.com/SigmaHQ/pySigma-backend-insightidr/compare/refs/tags/v0.1.9...v0.2.1

Ma27 committed:
nixos/tests/nextcloud: fix more issues related to redis tests

* Ensure that the redis cache is actually used in the "trivial" case
  (`with-postgresql-and-redis`)
* Test against all Nextcloud versions we've packaged
* Actually set a secret to make sure that the provided secret is
  properly read by Nextcloud.
* Add myself as maintainer to the secret-test to make sure that I don't
  miss any more changes like this that could break the functionality of
  that feature.

bobby285271 committed:
libgepub: 0.7.0 → 0.7.1

https://gitlab.gnome.org/GNOME/libgepub/-/compare/0.7.0...0.7.1

bobby285271 committed:
xfce.tumbler: enable EPUB thumbnailer support

Introduced in https://gitlab.xfce.org/xfce/tumbler/-/merge_requests/6

dotlambda committed:
rnote: 0.6.0 -> 0.7.0

Diff: https://github.com/flxzt/rnote/compare/v0.6.0...v0.7.0

Changelog: https://github.com/flxzt/rnote/releases/tag/v0.7.0

dotlambda committed:
libdeltachat: 1.116.0 -> 1.117.0

Diff: https://github.com/deltachat/deltachat-core-rust/compare/v1.116.0...v1.117.0

Changelog: https://github.com/deltachat/deltachat-core-rust/blob/v1.117.0/CHANGELOG.md

K900 committed:
buildMozillaMach: cherry-pick patches to fix ESR builds

apfelkuchen6 committed:
git-latexdiff: 1.3.0 -> 1.6.0

RaitoBezarius commented:
Can you detail more what is meant by zstd support and what does it bring concretely?

devusb committed:
sunshine: 0.19.1 -> 0.20.0

r-ryantm committed:
xml2rfc: 3.17.2 -> 3.17.3

r-ryantm committed:
codec2: 1.1.0 -> 1.1.1

Ma27 committed:
prometheus-postgres-exporter: 0.12.0 -> 0.12.1

ChangeLog: https://github.com/prometheus-community/postgres_exporter/releases/tag/v0.12.1

Ma27 committed:
prometheus-redis-exporter: 1.50.0 -> 1.51.0

ChangeLog: https://github.com/oliver006/redis_exporter/releases/tag/v1.51.0

r-ryantm committed:
grpc_cli: 1.55.1 -> 1.56.0

qjoly committed:
Subtitlr: 0.1.1 -> 0.2.0

mweinelt committed:
python310Packages.django-mysql: 4.9.0 -> 4.10.0

https://github.com/adamchainz/django-mysql/blob/4.10.0/docs/changelog.rst
https://github.com/adamchainz/django-mysql/compare/refs/tags/4.9.0...4.10.0

hughobrien committed:
iac: init at 0.1.7

r-ryantm committed:
rain: 1.4.0 -> 1.4.2

r-ryantm committed:
steampipe: 0.20.5 -> 0.20.6

alyssais committed:
crosvm: 113.0 -> 114.1

rrbutani commented:
> Can you detail more what is meant by zstd support and what does it bring concretely?

@RaitoBezarius This enables features like [`clang -gz=zstd`](https://github.com/llvm/llvm-project/commit/a0365abad8113d7e60294c5e581c24ec06ba037e) (compressing debug sections with zstd) and corresponding [support](https://reviews.llvm.org/D133548) in [`lld`](https://github.com/llvm/llvm-project/commit/449f2ca146dde0ef14c9837f71d4a0be92ca4843) and in the LLVM binutils.

r-ryantm committed:
spire: 1.6.4 -> 1.7.0

RaitoBezarius committed:
nixos/netdata: fix claim_token_file -> claimTokenFile

It was forgotten in afc6f0d93babfd795bd25f828e3ef3cdb145953d.

r-ryantm committed:
step-kms-plugin: 0.8.3 -> 0.9.0

rrbutani reviewed (changes requested):
Thanks for opening this PR! `zstd` support in LLVM is definitely something we overlooked.

---

A few comments:
  - `zstd` support in `clang` and `lld` did land in LLVM 16 but `zstd` was added to the LLVM support library in LLVM 15: https://github.com/llvm/llvm-project/commit/e939bf67e34037970192fa90ad22ab7628f5de33; can you amend this PR to backport this change to `llvmPackages_15` as well?
    + edit: will do in a follow up PR
  - [ ] it'd be nice to have some tests for this (i.e. a passthru test on `clang` that attempts to pass `-gz=zstd` if `enableZstd` is `true` on the underlying `llvm`) but I don't think it's a blocker for this PR
  - [ ] LLVM also has a [`LLVM_USE_STATIC_ZSTD` option](https://github.com/llvm/llvm-project/commit/fc1da043f4f9198303abd6f643cf23439115ce73) (added LLVM 16) that does not appear to be autodetected; we may want to pass this explicitly as part of the cmake flags when `stdenv.hostPlatform.isStatic` is `true` (though I am not sure if this is actually required...)

r-ryantm committed:
pico-sdk: 1.5.0 -> 1.5.1

r-ryantm committed:
plantuml: 1.2023.8 -> 1.2023.9

r-ryantm committed:
lxgw-neoxihei: 1.010 -> 1.101

kilianar committed:
portfolio: 0.63.1 -> 0.64.0

https://github.com/buchen/portfolio/releases/tag/0.64.0

sikmir committed:
lagrange: 1.16.3 → 1.16.4

r-ryantm committed:
helm-ls: 0.0.3 -> 0.0.4

figsoda committed:
bfc: 1.10.0 -> 1.11.0

Diff: https://github.com/Wilfred/bfc/compare/1.10.0...1.11.0

figsoda committed:
rsonpath: 0.4.0 -> 0.5.0

Diff: https://github.com/v0ldek/rsonpath/compare/v0.4.0...v0.5.0

Changelog: https://github.com/v0ldek/rsonpath/blob/v0.5.0/CHANGELOG.md

figsoda committed:
halp: 0.1.6 -> 0.1.7

Diff: https://github.com/orhun/halp/compare/v0.1.6...v0.1.7

Changelog: https://github.com/orhun/halp/blob/v0.1.7/CHANGELOG.md

HollowMan6 committed:
mdbook-pdf: 0.1.6 -> 0.1.7

Signed-off-by: Hollow Man <hollowman@opensuse.org>

pongo1231 committed:
distrobox: 1.4.2.1 -> 1.5.0

sikmir committed:
simplotask: 1.7.0 → 1.8.1

Artturin committed:
writers.makePythonWriter: fix interpreter executable

Aleksanaa committed:
marker: 2020.04.04.2 -> 2023.05.02

Aleksanaa committed:
marker: add aleksana as maintainer

r-ryantm committed:
tun2socks: 2.5.0 -> 2.5.1

markuskowa committed:
openmolcas: 23.02 -> 23.06

Artturin committed:
tests.stdenv.hooks.reproducible-builds: fix mistake

r-ryantm committed:
osdlyrics: 0.5.13 -> 0.5.14

r-ryantm committed:
doppler: 3.60.1 -> 3.61.0

Garmelon committed:
nixos/grafana: reformat

drupol committed:
phpactor: 2023.01.21 -> 2023.06.17

figsoda committed:
python310Packages.ruff-lsp: 0.0.27 -> 0.0.31

Changelog: https://github.com/astral-sh/ruff-lsp/releases/tag/v0.0.31

figsoda committed:
egglog: unstable-2023-05-22 -> unstable-2023-06-11

Diff: https://github.com/egraphs-good/egglog/compare/5242b50051c339d55009860d4dff80125fdcedfd...c7ef8b000caf7fa17f6127847db4b9c285c03f09

Garmelon committed:
nixos/grafana: update and add settings

I went through all categories that were already present in the grafana
module and added most options from the official docs at
https://grafana.com/docs/grafana/v9.5/setup-grafana/configure-grafana/

I also modified the descriptions of some existing options to match the
official docs more closely.

r-ryantm committed:
python310Packages.nunavut: 2.1.0 -> 2.1.1

r-ryantm committed:
python310Packages.dataclasses-json: 0.5.7 -> 0.5.9

wegank committed:
dbeaver: cleanup

r-ryantm committed:
hydroxide: 0.2.26 -> 0.2.27

Commit:
catppuccin-kde: fix install script

r-ryantm committed:
python310Packages.somajo: 2.2.3 -> 2.2.4

tejing1 committed:
vieb: 9.7.1 -> 10.0.0

r3n3gad3p3arl committed:
vimPlugins.scnvim: init at 2023-04-29

aaronjheng committed:
ssm-session-manager-plugin: refactor

aaronjheng committed:
kube-router: unpin go1.19

SaltyKitkat commented on pkgs/development/compilers/llvm/16/lldb/default.nix:
```diff
@@ -76,6 +77,7 @@ stdenv.mkDerivation (rec {
   buildInputs = [
     ncurses
     zlib
+    zstd
```
I remove both zlib and zstd in this file
and it seems that lldb can still be compiled with zlib and zstd support?
cmake found zlib and zstd anyway.

SaltyKitkat commented on pkgs/development/compilers/llvm/16/lldb/default.nix:
```diff
@@ -4,6 +4,7 @@
 , cmake
 , ninja
 , zlib
+, zstd
```
> Is there some functionality in `lldb` that requires `zstd` as a dep here?

Actually I don't know. I just search where `zlib` appears, and add `zstd`. LOL

r-ryantm committed:
circleci-cli: 0.1.26837 -> 0.1.26896

r-ryantm committed:
kapp: 0.57.0 -> 0.57.1

r-ryantm committed:
kind: 0.19.0 -> 0.20.0

figsoda committed:
vimPlugins: update

figsoda committed:
vimPlugins.nvim-treesitter: update grammars

r-ryantm committed:
kubefirst: 2.1.0 -> 2.1.4

r-ryantm committed:
hcloud: 1.34.1 -> 1.35.0

r-ryantm committed:
zincsearch: 0.4.5 -> 0.4.6

kirillrdy committed:
icu: 73.1 -> 73.2

marsam committed:
tflint: 0.46.1 -> 0.47.0

Diff: https://github.com/terraform-linters/tflint/compare/v0.46.1...v0.47.0

Changelog: https://github.com/terraform-linters/tflint/blob/v0.47.0/CHANGELOG.md

foo-dogsquared committed:
eartag: 0.4.0 -> 0.4.1

r-ryantm committed:
pulumi: 3.69.0 -> 3.71.0

r-ryantm committed:
ghorg: 1.9.5 -> 1.9.6

r-ryantm committed:
ugrep: 3.11.2 -> 3.12.1

r-ryantm committed:
pscale: 0.145.0 -> 0.147.0

r-ryantm committed:
krapslog: 0.5.1 -> 0.5.2

dit7ya committed:
codux: init at 15.4.4

r-ryantm committed:
nats-server: 2.9.17 -> 2.9.18

r-ryantm committed:
thc-hydra: 9.4 -> 9.5

r-ryantm committed:
lefthook: 1.4.1 -> 1.4.2

SaltyKitkat commented:
@rrbutani oops… backporting this to llvm15 seems cause lots of rebuild

should we open a separate pr for that?

r-ryantm committed:
buildkite-agent: 3.47.0 -> 3.48.0

fabaff committed:
python311Packages.asteval: 0.9.29 -> 0.9.30

Changelog: https://github.com/newville/asteval/releases/tag/0.9.30

fabaff committed:
python311Packages.sqltrie: 0.6.0 -> 0.7.0

Diff: https://github.com/iterative/sqltrie/compare/refs/tags/0.6.0...0.7.0

Changelog: https://github.com/iterative/sqltrie/releases/tag/0.7.0

fabaff committed:
python311Packages.pywerview: 0.5.0 -> 0.5.1

Diff: https://github.com/the-useless-one/pywerview/compare/refs/tags/v0.5.0...v0.5.1

Changelog: https://github.com/the-useless-one/pywerview/releases/tag/v0.5.1

r-ryantm committed:
trust-dns: 0.22.0 -> 0.22.1

WhiteBlackGoose committed:
avalonia-ilspy: added mainProgram attribute

fabaff committed:
python3Packages.django-leaflet: add changelog to meta

Azd325 committed:
vscode-extensions.charliermarsh.ruff: 2023.22.0 -> 2023.24.0

- update project urls
https://astral.sh/blog/announcing-astral-the-company-behind-ruff

r-ryantm committed:
python310Packages.json-rpc: 1.14.0 -> 1.15.0

bobby285271 committed:
nixos/pantheon: allow excluding pantheon-agent-geoclue2

Bash will raise syntax error for sessionCommands if sessionPath is empty.

vcunat committed:
gst_all_1.gst-plugins-rs: drop a test on aarch64-*

symphorien committed:
Revert "build-rust-{crate,package}: cleanups"

NickCao committed:
apcupsd: properly set configureFlags, fix cross compilation

NickCao committed:
lrs: set CC, fix cross compilation

r-ryantm committed:
phlare: 0.5.1 -> 0.6.1

symphorien committed:
poetry2nix overrides: readd ihatemoney

it was unduely removed in https://github.com/NixOS/nixpkgs/pull/225285

See https://github.com/NixOS/nixpkgs/pull/225285/files/3cc8b4477e175bb700022c147a59bd40e0de8161#diff-023a4624b5b2b252529285ae13a4827ca0343fc11a93630c0a6c87eb550b258a

kirillrdy committed:
python3Packages.angrcli: enable passing tests

r-ryantm committed:
snarkos: 2.0.2 -> 2.1.1

hellwolf committed:
owncloud-client: 4.0.0 -> 4.1.0

dasj19 committed:
giara: 1.0.1 -> 1.1.0

r-ryantm committed:
conftest: 0.42.1 -> 0.43.1

r-ryantm committed:
sentry-cli: 2.18.1 -> 2.19.0

marsam committed:
sentry-cli: 2.19.0 -> 2.19.1

Diff: https://github.com/getsentry/sentry-cli/compare/2.19.0...2.19.1

Changelog: https://github.com/getsentry/sentry-cli/raw/2.19.1/CHANGELOG.md

Commit:
pot: 1.1.0 -> 1.6.1

matthiasbeyer committed:
cargo-public-api: 0.31.1 -> 0.31.2

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

r-ryantm committed:
microdnf: 3.9.1 -> 3.10.0

matthiasbeyer committed:
zellij: 0.36.0 -> 0.37.0

Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

thiagokokada committed:
wl-clip-persist: init at 0.3.1

markuskowa committed:
snapper: 0.10.4 -> 0.10.5

markuskowa committed:
nixos/snapper/test: fix, make compliant with new config scheme

figsoda committed:
vimPlugins.sniprun: 1.3.3 -> 1.3.4

Diff: https://github.com/michaelb/sniprun/compare/v1.3.3...v1.3.4

Changelog: https://github.com/michaelb/sniprun/blob/v1.3.4/CHANGELOG.md

figsoda committed:
typical: 0.9.5 -> 0.9.6

Diff: https://github.com/stepchowfun/typical/compare/v0.9.5...v0.9.6

Changelog: https://github.com/stepchowfun/typical/blob/v0.9.6/CHANGELOG.md

figsoda committed:
gtree: 1.7.48 -> 1.7.50

Diff: https://github.com/ddddddO/gtree/compare/v1.7.48...v1.7.50

Changelog: https://github.com/ddddddO/gtree/releases/tag/v1.7.50

figsoda committed:
rust-script: 0.28.0 -> 0.29.0

Diff: https://github.com/fornwall/rust-script/compare/0.28.0...0.29.0

Changelog: https://github.com/fornwall/rust-script/releases/tag/0.29.0

fabaff committed:
python311Packages.pypoolstation: 0.4.9 -> 0.5.1

figsoda committed:
oculante: 0.6.64 -> 0.6.65

Diff: https://github.com/woelper/oculante/compare/0.6.64...0.6.65

Changelog: https://github.com/woelper/oculante/blob/0.6.65/CHANGELOG.md

SuperSandro2000 committed:
shfmt: 3.6.0 -> 3.7.0

Diff: https://github.com/mvdan/sh/compare/v3.6.0...v3.7.0

apfelkuchen6 committed:
licenses: add Knuth license

rrbutani commented:
> backporting this to llvm15 seems cause lots of rebuild
> 
> should we open a separate pr for that?

Sure! If/when you open that PR, be sure to target `staging`.

r-ryantm committed:
python311Packages.cssutils: 2.7.0 -> 2.7.1

r-ryantm committed:
i2p: 2.2.0 -> 2.2.1

nullishamy committed:
maintainer: update email

This changes my email to be up to date.

yuyuyureka committed:
scrcpy: use adb from android-tools

android-tools is built from source and available on aarch64-linux, while platform tools isn't

apfelkuchen6 committed:
licenses: add LPPL-1.0

apfelkuchen6 committed:
licenses: add LPPL-1.3a

apfelkuchen6 committed:
licenses: add CC-BY-1.0

apfelkuchen6 committed:
licenses: add CC-BY-SA-1.0

apfelkuchen6 committed:
licenses: add CC-BY-SA-2.0

apfelkuchen6 committed:
texlive.tlpdb.nix: extract licensing information

apfelkuchen6 committed:
texlive: add license to texlive packages

apfelkuchen6 committed:
texlive.combine: expose licensing information of combined packages

drupol committed:
vscode-extensions.devsense.composer-php-vscode: 1.33.13032 -> 1.34.13295

drupol committed:
vscode-extensions.devsense.profiler-php-vscode: 1.33.13032 -> 1.34.13295

drupol committed:
vscode-extensions.devsense.phptools-vscode: 1.33.13032 -> 1.34.13295

r-ryantm committed:
python310Packages.trezor: 0.13.6 -> 0.13.7

SuperSandro2000 committed:
python310Packages.pytest-testinfra: fix formatting

r-ryantm committed:
python310Packages.django-cleanup: 7.0.0 -> 8.0.0

lopsided98 committed:
urdfdom: 3.1.0 -> 3.1.1

erdnaxe committed:
suricata: 6.0.12 -> 6.0.13

OPNA2608 committed:
libopenmpt: 0.7.1 -> 0.7.2

r-ryantm committed:
sarasa-gothic: 0.41.0 -> 0.41.2

Commit:
lib/systems: remove redundant test from selectEmulator

Commit eef4bbd82f4c2 changed the conditional in selectEmulator from
`isCompatible` (which examines only the CPU, rather than the entire
platform) to `canExecute`.  This made the first conjunct redundant.
Let's drop the redundant part.

https://github.com/NixOS/nixpkgs/pull/238331#discussion_r1233277119

r-ryantm committed:
android-udev-rules: 20230303 -> 20230614

OPNA2608 committed:
vgmtools: unstable-2023-04-17 -> unstable-2023-05-04

r-ryantm committed:
glooctl: 1.14.7 -> 1.14.8

apfelkuchen6 committed:
licenses: add Artistic-1.0-cl8

felschr committed:
python310Packages.drf-nested-routers: 0.93.3 -> 0.93.4 (#238077)

apfelkuchen6 committed:
licenses: add GFSL

apfelkuchen6 committed:
licenses: add GFL

r-ryantm committed:
konstraint: 0.29.0 -> 0.29.1

r-ryantm committed:
python310Packages.annoy: 1.17.2 -> 1.17.3

SaltyKitkat commented on pkgs/development/compilers/llvm/16/llvm/default.nix:
```diff
@@ -96,7 +97,7 @@ in
     ++ optional enablePFM libpfm; # exegesis
 
-  propagatedBuildInputs = [ ncurses zlib ];
+  propagatedBuildInputs = [ ncurses zlib zstd ];
```
> Can we gate this on an `enableZstd` input to the package (defaulted to `true)?

I'm a little confused here.

- Why we need a feature gate here? It seems that no zstd support brings no benefits from my perspective?

- I know nothing about tests in nixpkgs. Maybe I need some help about the tests.

RaitoBezarius replied:
Feature gate is useful to control closure sizes and sometimes when something make a special variant incompatible, e.g. static builds, musl builds, cross compilation, etc.

RaitoBezarius replied:
If we passthru it, we can recognize whether any LLVM, Clang, LLD at hand is able to do zstd or not and we can decide whether our test harness will attempt to decompress stuff using zstd. Implementing the passthru is the only thing needed here at the moment AFAIK.

rrbutani commented:
Now that #236420 has reached `master` I think we'll need to retarget this PR to `staging`; there are too many LLVM 16 rdeps (via Rust 1.70).

@SaltyKitkat would you mind adding back in the LLVM 15 changes and retargeting this PR? (instructions for retargeting to `staging` are [here](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging))

SaltyKitkat committed:
llvm 16: zstd: add passthru and feature gate

SaltyKitkat commented:
Oops…
